config/schema.json

```diff
@@ -0,0 +1,299 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://hashicorp.com/schemas/consul-ecs",
+  "title": "Consul ECS Configuration",
+  "description": "Configuration for the consul-ecs binary",
+  "type": "object",
+  "properties": {
+    "aclTokenSecret": {
+      "description": "Configure the ACL token secret provider",
+      "type": "object",
+      "properties": {
+        "provider": {
+          "type": "string",
+          "enum": [
+            "secrets-manager"
+          ]
+        },
+        "configuration": {
+          "type": "object",
+          "properties": {
+            "prefix": {
+              "type": "string"
+            },
+            "consulClientTokenSecretArn": {
+              "type": "string"
+            }
+          },
+          "required": [
+            "prefix",
+            "consulClientTokenSecretArn"
+          ],
+          "additionalProperties": false
+        }
+      },
+      "required": [
+        "provider",
+        "configuration"
+      ],
+      "additionalProperties": false
+    },
+    "mesh": {
+      "description": "Configure the Consul mesh",
+      "type": "object",
+      "properties": {
+        "bootstrapDir": {
+          "type": "string"
+        },
+        "healthSyncContainers": {
```

Comment:
What do you think about adding descriptions to the fields that don't come from Consul types? Some examples are `healthSyncContainers` and the fields in `aclTokenSecret`. This can be done in another PR though.